tpot/export_utils.py

```diff
@@ -393,6 +394,24 @@ def replace_function_calls(pipeline_list):
 
 {OUTPUT_DF}['gbc{OPERATOR_NUM}-classification'] = gbc{OPERATOR_NUM}.predict({OUTPUT_DF}.drop('class', axis=1).values)
 """.format(OUTPUT_DF=result_name, OPERATOR_NUM=operator_num, LEARNING_RATE=learning_rate, MAX_FEATURES=max_features, MIN_WEIGHT=min_weight)
+        
+        elif operator_name == '_xg_boosting':
+            
+            learning_rate = min(1., max(float(operator[3]), 0.0001))
+            subsample = min(1., max(0.1, float(operator[4])))            
+            min_child_weight = min(100, max(0, int(operator[5])))
+            max_depth = min(100, max(1, int(operator[6])))            
```

Comment:
I've used learning_rate, subsample and min_child_weight as these are the counterparts of     learning_rate, max_features and min_weight present in _gradient_boosting. Also, from my experience, max_depth plays a very important role to prevent overfitting (https://github.com/dmlc/xgboost/blob/master/doc/how_to/param_tuning.md).

I got the value caps from https://github.com/dmlc/xgboost/blob/master/doc/parameter.md (learning_rate (eta) = [0,1], subsample = (0, 1], min_child_weight = [0, inifity], max_depth = [1, inifity]).
Base on these I used some common sense for:
-  subsample: 0 samples per tree makes no sense for me
- max_depth : a depth of 0 means no tree (for what I understand), and a tree that is too deep will surely overfit